tests/integration/high_availability/application_charm/src/charm.py

```diff
@@ -0,0 +1,192 @@
+#!/usr/bin/env python3
+# Copyright 2022 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Application charm that connects to database charms.
+
+This charm is meant to be used only for testing
+high availability of the MySQL charm.
+"""
+
+import logging
+import subprocess
+from typing import Optional
+
+from charms.data_platform_libs.v0.database_requires import DatabaseRequires
+from connector import MySQLConnector
+from ops.charm import ActionEvent, CharmBase
+from ops.framework import StoredState
+from ops.main import main
+from ops.model import ActiveStatus, WaitingStatus
+from tenacity import RetryError, Retrying, stop_after_delay, wait_fixed
+
+logger = logging.getLogger(__name__)
+
+DATABASE_NAME = "continuous_writes_database"
+TABLE_NAME = "data"
+
+
+class ContinuousWritesApplication(CharmBase):
+    """Application charm that continuously writes to MySQL."""
+
+    _stored = StoredState()
```

Comment:
I don't remember exactly on which conditions this was not advised, but if any issue you could still resort to a peers relation data-bag

Reply:
great call! changed to use app peer databag in 0986dce467170ce6ac07013fda91adecffcf63ab